cli/file_watcher.rs

```diff
@@ -0,0 +1,80 @@
+use crate::colors;
+use deno_core::ErrBox;
+use futures::stream::StreamExt;
+use futures::Future;
+use notify::event::Event as NotifyEvent;
+use notify::event::EventKind;
+use notify::Error as NotifyError;
+use notify::RecommendedWatcher;
+use notify::RecursiveMode;
+use notify::Watcher;
+use std::path::PathBuf;
+use std::pin::Pin;
+use tokio::select;
+use tokio::sync::mpsc;
+
+// TODO(bartlomieju): rename
+type WatchFuture =
+  Pin<Box<dyn Future<Output = std::result::Result<(), deno_core::ErrBox>>>>;
+
+async fn error_handler(watch_future: WatchFuture) {
+  let result = watch_future.await;
+  if let Err(err) = result {
+    let msg = format!("{}: {}", colors::red_bold("error"), err.to_string(),);
+    eprintln!("{}", msg);
+  }
+}
+
+pub async fn watch_func<F>(
+  watch_paths: &[PathBuf],
+  closure: F,
+) -> Result<(), ErrBox>
+where
+  F: Fn() -> WatchFuture,
+{
+  loop {
+    let func = error_handler(closure());
+    let mut is_file_changed = false;
+    select! {
+      _ = file_watcher(watch_paths) => {
+          is_file_changed = true;
+          println!("File change detected! Restarting!");
```

Comment:
I think we should add some colors to this message. It gets lost easily otherwise. Maybe `${blue("Watcher")} Restarting due to file change.`, and `${blue("Watcher")} Process terminated. Waiting for file change...`

Also this should probably respect `--quiet`.